File: leetcode/Medium/633. Sum of Square Numbers/633.java
class Solution {
    public boolean judgeSquareSum(int c) {
        long l = 0, r = (long)Math.sqrt(c);

        while(l<=r){
            if(l*l + r*r == c) return true;
            else if (l*l + r*r < c) l++;
            else r--;
        }
        return false;
        
    }
}

// TC: O(sqrt(n))
// SC: O(1)

File: leetcode/Medium/633. Sum of Square Numbers/633.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        long long l = 0, r = static_cast<long long>(sqrt(c));
        while(l <= r) {
            if(l * l + r * r == c) return true;
            else if(l * l + r * r > c) r--;
            else l++;
        }

        return false;
    }
};

// TC: O(sqrt(c))
// SC: O(1)